packages/blade/src/components/Drawer/__tests__/Drawer.web.test.tsx

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import userEvents from '@testing-library/user-event';
+import { waitFor } from '@testing-library/react';
+import React from 'react';
+import type { DrawerProps } from '../';
+import { Drawer, DrawerBody, DrawerHeader } from '../';
+import renderWithTheme from '~utils/testing/renderWithTheme.web';
+import { Badge } from '~components/Badge';
+import { Button } from '~components/Button';
+import { Text } from '~components/Typography';
+import { AnnouncementIcon, DownloadIcon } from '~components/Icons';
+
+const BasicDrawer = (props: Partial<DrawerProps>): React.ReactElement => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  return (
+    <>
+      <Drawer
+        {...props}
+        isOpen={isOpen}
+        onDismiss={() => {
+          setIsOpen(false);
+          props.onDismiss?.();
+        }}
+        accessibilityLabel="Test Drawer"
+      >
+        <DrawerHeader title="Drawer" />
+        <DrawerBody>
+          <Text>Test Content</Text>
+        </DrawerBody>
+      </Drawer>
+      <Button onClick={() => setIsOpen(true)}>Toggle Drawer</Button>
+    </>
+  );
+};
+
+describe('Drawer', () => {
+  it('renders a Drawer', () => {
+    const { getByRole } = renderWithTheme(
+      <Drawer isOpen={true} onDismiss={() => {}} accessibilityLabel="Test Drawer">
+        <DrawerHeader
+          leading={<AnnouncementIcon size="large" />}
+          title="Address Details"
+          subtitle="Saving addresses will improve your checkout experience"
+          trailing={<Button icon={DownloadIcon} />}
+          titleSuffix={<Badge color="positive">NEW</Badge>}
+        />
+        <DrawerBody>
+          <Text>Test Content</Text>
+        </DrawerBody>
+      </Drawer>,
+    );
+    expect(getByRole('dialog')).toMatchSnapshot();
+  });
+
+  it('should open and close drawer', async () => {
+    const user = userEvents.setup();
+
+    const onDismiss = jest.fn();
+
+    const { getByRole, queryByRole } = renderWithTheme(<BasicDrawer onDismiss={onDismiss} />);
+    await user.click(getByRole('button', { name: 'Toggle Drawer' }));
+    await waitFor(() => expect(getByRole('dialog')).toBeVisible());
+    await user.click(getByRole('button', { name: 'Close' }));
+    await waitFor(() => expect(queryByRole('dialog')).not.toBeInTheDocument());
+    expect(onDismiss).toHaveBeenCalledTimes(1);
+  });
+
+  it('should dismiss on overlay click', async () => {
```

Comment:
I don't think we need this, if this case is already covered via interaction tests.

Reply:
I wanted to keep some basic functionality tests on unit test as well since interaction tests we can miss to run. So unit test only covers open close handlers and critical dismiss behaviour